src/components/accounts/SetupImplicit.js

```diff
@@ -0,0 +1,289 @@
+import React, { Component, Fragment } from 'react'
+
+import * as nearApiJs from 'near-api-js' 
+import BN from 'bn.js'
+
+import { withRouter, Route } from 'react-router-dom'
+import { connect } from 'react-redux'
+import { Translate } from 'react-localize-redux'
+
+import { redirectToApp, handleAddAccessKeySeedPhrase, clearAlert, refreshAccount, checkCanEnableTwoFactor, checkIsNew, handleCreateAccountWithSeedPhrase } from '../../actions/account'
+import { generateSeedPhrase } from 'near-seed-phrase'
+import SetupSeedPhraseVerify from './SetupSeedPhraseVerify'
+import SetupSeedPhraseForm from './SetupSeedPhraseForm'
+import copyText from '../../utils/copyText'
+import isMobile from '../../utils/isMobile'
+import { Snackbar, snackbarDuration } from '../common/Snackbar'
+import Container from '../common/styled/Container.css'
+import { KeyPair } from 'near-api-js'
+import FormButton from '../common/FormButton'
+import IconMCopy from '../../images/IconMCopy'
+const NETWORK_ID = process.env.REACT_APP_NETWORK_ID || 'default'
+const NODE_URL = process.env.REACT_APP_NODE_URL || 'https://rpc.nearprotocol.com'
+
+const IMPLICIT_ACCOUNT_KEY = '__IMPLICIT_ACCOUNT_KEY'
+
+class SetupImplicit extends Component {
+    state = {
+        seedPhrase: '',
+        enterWord: '',
+        wordId: null,
+        requestStatus: null,
+        successSnackbar: false,
+        sending: false,
+        snackBarMessage: 'setupSeedPhrase.snackbarCopySuccess'
+    }
+
+    async checkBalance() {
+        const recoveryKeyPair = JSON.parse(localStorage.getItem(IMPLICIT_ACCOUNT_KEY) || '{}')
+        this.setState({ sending: true })
+        const account = new nearApiJs.Account(this.connection, recoveryKeyPair.implicitAccountId)
+        try {
+            const state = await account.state()
+            if (new BN(state.amount).gte(new BN('1000000000000000000000000', 10))) {
+                window.removeEventListener('beforeunload')
+                window.location.href = `${window.location.origin}/recover-with-link/${recoveryKeyPair.implicitAccountId}/${recoveryKeyPair.seedPhrase}`
+            }
+        } catch (e) {
```

Comment:
see another comment – I think we really should just make it easier to check whether account exists (have a helper function)